src/os/bluestore/bluefs_types.h

```diff
@@ -242,6 +279,21 @@ struct bluefs_transaction_t {
     encode((__u8)OP_FILE_UPDATE, op_bl);
     encode(file, op_bl);
   }
+  void op_file_update_inc(const bluefs_fnode_t& file) {
+    using ceph::encode;
+    encode((__u8)OP_FILE_UPDATE_INC, op_bl);
+    bluefs_fnode_delta_t delta;
+    delta.ino = file.ino;
+    delta.size = file.size;
+    delta.mtime = file.mtime;
+    uint64_t new_extents_size = 0;
+    for (const auto &p : file.new_extents) {
```

Comment:
nit: new_extents_size can be tracked at bluefs_fnode_t on adding new extent, no need to reiterate over the new_extents then.